src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -734,4 +811,18 @@ protected function isLocalResource($sResource)
 
         return $bIsLocal;
     }
+
+    private function hasResourceCollectionWriteProcessRunning(): bool
+    {
+        return file_exists($this->assetPath.'/lock.tmp');
+    }
+
+    private function setsResourceCollectionWriteProcessRunning(bool $state): bool
+    {
+        if (true === $state) {
+            return false === $this->hasResourceCollectionWriteProcessRunning() && touch($this->assetPath.'/lock.tmp');
```

Comment:
It's a bit surprising to have a set method do these things and also the hierarchy levels of the code are mixed ("check for file" is another helper method and the "create" isn't).

So for simplification / clarification: Why not have three methods:
isResourceCollectionProcessActive, startResourceCollectionProcess and finishResourceCollectionProcess?

Reply:
you are right - we should be checking for the existence of the file and not call `hasResourceCollectionWriteProcessRunning`. This way the method has only one level of abstraction and it becomes much clearer what is going on.

I changed the method along those lines.